src/main/java/com/ays/assignment/model/dto/request/AssignmentListRequest.java

```diff
@@ -58,21 +74,49 @@ public boolean isSortPropertyAccepted() {
         return this.isPropertyAccepted(acceptedFilterFields);
     }
 
+
     /**
-     * Converts the request into a JPA Specification that filters assignments based on the specified status,
-     * if it is provided.
+     * Converts the request into a JPA Specification that filters assignments based on the specified
+     * statuses and phoneNumber, if they are provided.
      *
      * @param clazz the class type of the specification.
      * @return the generated JPA Specification based on the request filters.
      */
     @Override
     public <C> Specification<C> toSpecification(Class<C> clazz) {
+
+        if (this.filter == null) {
+            return Specification.allOf();
+        }
+
         Specification<C> specification = Specification.where(null);
-        if (status != null) {
-            specification = specification.and((root, query, builder) ->
-                    builder.equal(root.get("status"), status));
+
+        if (this.filter.phoneNumber != null) {
+
+            if (this.filter.phoneNumber.getLineNumber() != null && !this.filter.phoneNumber.getLineNumber().isEmpty()) {
+                Specification<C> lineNumberSpecification = (root, query, criteriaBuilder) ->
+                        criteriaBuilder.equal(root.get("lineNumber"), this.filter.phoneNumber.getLineNumber());
+
+                specification = specification.and(lineNumberSpecification);
+            }
+
+            if (this.filter.phoneNumber.getCountryCode() != null && !this.filter.phoneNumber.getCountryCode().isEmpty()) {
```

Comment:
Sadece `!this.filter.phoneNumber.getCountryCode().isBlank()` şeklinde tanımlasak buradaki kontrolleri karşılamıyor mu?

Reply:
PhoneNumber'da Digit anotasyonu olduğu için "" ve "  " değerlerinde zaten "MUST BE NUMBER" hatası dönülüyorç
Bu durumda sadece null kontrolü yapmak yeterli. Değiştirdim. @agitrubard